nomad/server.go

```diff
@@ -1134,8 +1134,14 @@ func (s *Server) setupConsul(consulConfigEntries consul.ConfigAPI, consulACLs co
 
 // setupVaultClient is used to set up the Vault API client.
 func (s *Server) setupVaultClient() error {
+	vconfig := s.config.VaultConfig
+	if vconfig != nil && vconfig.UseIdentity != nil && *vconfig.UseIdentity {
+		s.vault = &NoopVault{}
+		return nil
+	}
```

Comment:
Good catch on this, and it makes an obvious bit of code to tear out for 1.9.

Reply:
Yeah, at first I had `nomad.vaultClient` to be `nil` when `use_identity = true`, but that created a bunch of `nil` pointer exceptions 😅 

Having a no-op implementation feels safer and easier to remove in the future.